go/vt/vtgate/evalengine/fn_string.go

```diff
@@ -199,6 +205,77 @@ func (call *builtinLength) compile(c *compiler) (ctype, error) {
 	return c.compileFn_length(call.Arguments[0], c.asm.Fn_LENGTH)
 }
 
+func instrIndex(str *evalBytes, substr *evalBytes) int64 {
+	// Case sensitive if one of the strings is binary string
+	if !str.isBinary() && !substr.isBinary() {
+		str.bytes = bytes.ToLower(str.bytes)
+		substr.bytes = bytes.ToLower(substr.bytes)
+	}
+
+	pos := bytes.Index(str.bytes, substr.bytes) + 1
```

Comment:
There's a reason why `INSTR` is not implemented yet, and that is that we need to add functionality to the `go/mysql/collations` package to do the searching for a substring. This implementation here is wrong and doesn't consider collations correctly. It assumes that it's all only ASCII by the `ToLower` and doesn't handle non ASCII characters this way, nor does it handle any of the non-Unicode encodings here.